pep-0728.rst

```diff
@@ -0,0 +1,322 @@
+PEP: 728
+Title: Inner Fields Annotations
+Author: Nir Schulman <narsssx@gmail.com>
+Sponsor: Jelle Zijlstra <jelle.zijlstra@gmail.com>
```

Comment:
ping @JelleZijlstra for confirmation of sponsorship

Reply:
Sorry, but I am not sponsoring this PEP at the moment. I already have four open PEPs (696, 702, 724, 727), that feels like too much. I can think about sponsoring more when a few of the others are done.

Reply:
I see. In that case I'll wait and in the meantime address the rest of the comments (and in general try to improve the idea).

@AA-Turner Does this mean this PR should be closed for now or should it stay open with an appropriate label?

Reply:
@UltimateLobster I would suggest that you revisit the Discourse thread and see if another core developer interested in typing would sponsor this proposal. I'll close this PR for now so that we're aware that the PEP number isn't reserved, though still happy to discuss any of the comments I made in review.

A

Reply:
Just to set expectations, I don't know whether I'd be willing to sponsor the PEP if I had fewer open PEPs; I'd have to think more about whether I think this is a good direction to go into. I'd also prefer to see more evidence of the feasibility of the idea (e.g., buy-in from type checker maintainers, ideally a prototype implementation).